From 8668b3e1efe7dd858daf22e36290a3874a544863 Mon Sep 17 00:00:00 2001
From: "Frederic Pierret (Epitre)" <frederic.epitre@orange.fr>
Date: Tue, 19 Sep 2017 16:59:40 +0200
Subject: [PATCH 12/30] anaconda: use kernel-install instead of grubby to
 regenerate initrd/grub.conf

Since we have own hook there, it properly handles Xen. This means we no longer need post scripts in kickstart for that.
---
 pyanaconda/payload/__init__.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/pyanaconda/payload/__init__.py b/pyanaconda/payload/__init__.py
index 7e78145..dc46a32 100644
--- a/pyanaconda/payload/__init__.py
+++ b/pyanaconda/payload/__init__.py
@@ -657,7 +657,7 @@ class Payload(object):
                 #           prevent boot on some systems
 
     def recreateInitrds(self):
-        """Recreate the initrds by calling new-kernel-pkg
+        """Recreate the initrds by calling kernel-install
 
         This needs to be done after all configuration files have been
         written, since dracut depends on some of them.
@@ -671,9 +671,8 @@ class Payload(object):
         for kernel in self.kernelVersionList:
             log.info("recreating initrd for %s", kernel)
             if not flags.imageInstall:
-                iutil.execInSysroot("new-kernel-pkg",
-                                    ["--mkinitrd", "--dracut",
-                                    "--depmod", "--update", kernel])
+                iutil.execInSysroot("kernel-install",
+                                    ["add", kernel, "/boot/vmlinuz-%s" % kernel])
             else:
                 # hostonly is not sensible for disk image installations
                 # using /dev/disk/by-uuid/ is necessary due to disk image naming
-- 
2.9.5

